pkg/scheduler/apis/config/types.go

```diff
@@ -37,6 +40,22 @@ const (
 	SchedulerDefaultProviderName = "DefaultProvider"
 )
 
+// The following strings are duplicated here due to avoid dependency to pkg/scheduler/framework/...
```

Comment:
Can't use the following consts from ".../framework/" since that would generate a new dependency.

Reply:
we don't need any of them now.

Reply:
pkg/scheduler/apis/config/validation still needs `MaxNodeScore`. I updated the code to only duplicate `MaxNodeScore` in validation package.

Reply:
I don't think we should define ```MaxNodeScore``` in multiple places, we recently updated the score range to [0, 100]: https://github.com/kubernetes/kubernetes/pull/83522#event-2707651154


Here is the current situation:
1) [```MaxNodeScore```](https://github.com/kubernetes/kubernetes/blob/d6eba1df1466b72848255a7b6b3a9210d4ce6eca/pkg/scheduler/framework/v1alpha1/interface.go#L77) is the score for internal priorities/Score plugins
2) [```MaxExtenderScore```](https://github.com/kubernetes/kubernetes/blob/d6eba1df1466b72848255a7b6b3a9210d4ce6eca/pkg/scheduler/apis/extender/v1/types.go#L29) is the max score we expect from extenders.
3) [```CustomPriorityMaxScore```](https://github.com/kubernetes/kubernetes/blob/d6eba1df1466b72848255a7b6b3a9210d4ce6eca/pkg/scheduler/api/types.go#L32
) is the max priority we expect to see in [custom priority](https://github.com/kubernetes/kubernetes/blob/d6eba1df1466b72848255a7b6b3a9210d4ce6eca/pkg/scheduler/api/types.go#L154) (we should rename it to ```MaxCustomPriorityScore``` now that I noticed it is not consistent with the other two)


The validation logic seems to be validating the Weight, while MaxWeight right now depends on MaxNodeScore, I don't think we should duplicate the MaxNodeScore definition because of this, it is very reasonable to move MaxWeight to here from framework/interface.go, and define it as a static number, 1,000,000 for example (like what we do for critical priority).

Reply:
I agree we should move MaxWeight to `pkg/apis/...`, and as `pkg/apis` shouldn't depend on `framework` pkg, so let's put 1,000,000 here.

And I've also renamed `CustomPriorityScore` to `MaxCustomPriorityScore`. Please take a look at the latest commit.